src/pages/TransTable.jsx

```diff
@@ -29,25 +31,72 @@ const TransTable = () => {
-          {
-              sum = sum + parseFloat(transcation.amount)
-          }
-        });
-        setMonthSpent(sum.toFixed(2))
-    },[transactionList])
+    const [monthlyIncome, setMonthlyIncome] = useState('N/A');
+    const [surveyDataState, setSurveyDataState] = useState({});
+    const [savingsGoal, setSavingsGoal] = useState('N/A');
+    const [moneySaved, setMoneySaved] = useState('N/A');
+    const [totalSpent, setTotalSpent] = useState(0);
+    const [savingsPercentage, setSavingsPercentage] = useState(0);
+    const [moneyNeededToReachGoal, setMoneyNeededToReachGoal] = useState(0);
+
+    useEffect(() => {
+        if (currentUser) {
+          const currentUserUid = currentUser.uid;
+    
+          // Fetch transactions and survey data in parallel
+          Promise.all([
+            getTransactionFromDB(currentUserUid),
+            fetchSurveyData(currentUserUid)
+          ])
+          .then(([transactions, surveyData]) => {
+            setTransactionList(transactions);
+            
+            // Calculate the sum of all transaction amounts for the current month
+            const currentYear = new Date().getFullYear().toString();
+            const currentMonth = (new Date().getMonth() + 1).toString().padStart(2, '0');
+            const monthlySpent = transactions.reduce((sum, { date, amount }) => {
+              if (date.startsWith(`${currentYear}-${currentMonth}`)) {
+                return sum + parseFloat(amount);
+              }
+              return sum;
+            }, 0);
+            setMonthSpent(monthlySpent.toFixed(2));
+    
+            // Set survey data state
+            const userDataKey = Object.keys(surveyData)[0];
+            const userSurveyData = surveyData[userDataKey];
+            if (userSurveyData) {
+                console.log("USER SURVEY DATA:", userSurveyData);
+                if (userSurveyData.monthlyIncome) {
+                    setMonthlyIncome(userSurveyData.monthlyIncome);
+                }
+                if (userSurveyData.savingsGoal) {
+                    setSavingsGoal(parseFloat(userSurveyData.savingsGoal));
+                }
+                if (userSurveyData.moneySaved) {
+                    setMoneySaved(parseFloat(userSurveyData.moneySaved));
+                }
+
+                if (userSurveyData.moneySaved && userSurveyData.savingsGoal) {
+                    const savingsPercent = (parseFloat(userSurveyData.moneySaved) / parseFloat(userSurveyData.savingsGoal)) * 100;
+                    const difference = parseFloat(userSurveyData.savingsGoal) - parseFloat(userSurveyData.moneySaved);
+                    setSavingsPercentage(savingsPercent);
+                    setMoneyNeededToReachGoal(difference);
+                  }
+            }
+          })
+          .catch(error => {
+            console.error('Error fetching data:', error);
+          });
+        }
+      }, [currentUser]); // Dependencies array contains only currentUser    
```

Comment:
The monthly spending totals do not appear to update automatically when users add new transactions.  I think this is because in your useEffect only checks to see if the current user has changed, and does not detect if the transaction list has changed to update the monthly spending totals. for example you may need to write  useEffect(() => {....}, [currentUser, transactionList])

Reply:
Yeah typically you need to refresh. I can add something to make it automatically refresh when they add new transactions.